### DISCUSSION TITLE
NextAuth Environment Variables

### DISCUSSION THREAD

**Discussion by LaCocoRoco** [nextauthjs/next-auth#4357] [Help]
### Question 💬

I have some problems with the environment variables for the next-auth application. NextJS has the possibility to setup environment variables in the [next.config.js](https://nextjs.org/docs/api-reference/next.config.js/environment-variables).

```typescript
// next.config.mjs

const nextConfig = {
  reactStrictMode: true,
  env: {
    PROVIDER_CLIENT_ID: 'PROVIDER_CLIENT_ID',
    PROVIDER_CLIENT_SECRET: 'PROVIDER_CLIENT_SECRET',
    NEXTAUTH_SECRET: 'NEXTAUTH_SECRET',
    NEXTAUTH_URL: 'hNEXTAUTH_URL',
    NEXTAUTH_URL_INTERNAL: 'NEXTAUTH_URL_INTERNAL'
  }
}
```

But for some reason next-auth only loads the NEXTAUTH_URL, NEXTAUTH_URL_INTERNAL and NEXTAUTH_SECRET when i place them in an **.env** file. Is is possible to fix the SECRET issues by placing it into the NextAuth option.

```typescript
export default NextAuth({
  secret: process.env.NEXTAUTH_SECRET,
  providers: [
    Provider({
      clientId: process.env.PROVIDER_CLIENT_ID,
      clientSecret: process.env.PROVIDER_CLIENT_SECRET
    })
  ],
})

```

I couldn't find any solution to pass the URL and URL_INTERNAL as option or property. It was also not possible to load the environment variables via [npm config package](https://www.npmjs.com/package/config). Do i miss something or i am realy forced to use the environment variables?

### How to reproduce ☕️

1. Initialize a new Next Auth Sample Project.
2. Remove the default .env file.
3. Create **next.config.mjs** at project root level and replace content like descriped in the Question.

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by florMignini** [comment]
Hi, I hope is not too late, maybe it can be a simple solution to your problem
https://www.coderrocketfuel.com/article/add-environment-variables-to-a-next-js-application
